log_parser/event_pattern.py

```diff
@@ -29,6 +33,8 @@ def matches_pattern_at_index(self, message: str, index: int = 0) -> bool:
         Raises:
             IndexError: If the index is out of range for the compiled pattern list
         """
-        if not 0 <= index < len(self._compiled_patterns):
-            raise IndexError("Index out of range for the compiled pattern list.")
-        return bool(self._compiled_patterns[index].search(message))
+        if not (0 <= index < len(self.sequence)):
```

Comment:
In that case you can even flip the `if` condition as both solutions are "early return". Up to you